File: LeetCode/[M] - 560. Subarray Sum Equals K/solution.go
package main

func subarraySum(nums []int, k int) int {
	m := make(map[int]int)
	m[0] = 1
	sum := 0
	res := 0
	for _, num := range nums {
		sum += num
		if cnt, ok := m[sum-k]; ok {
			res += cnt
		}
		m[sum]++
	}
	return res
}

File: LeetCode/[M] - 560. Subarray Sum Equals K/solution.py
from typing import List


class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        sums = {}
        sums[0] = 1
        totalSum, res = 0, 0
        for num in nums:
            totalSum += num
            if totalSum - k in sums:
                res += sums[totalSum - k]
            sums[totalSum] = sums.get(totalSum, 0) + 1
        return res
